test-app/bundles/educandu-page-template.js

```diff
@@ -0,0 +1,203 @@
+          <React.Fragment key={lang.code}>
+            {index !== 0 && <span>/</span>}
+            <Button type="link" size="small" onClick={() => i18n.changeLanguage(lang.code)}>
+              <CountryFlagAndName code={lang.flag} name={lang.name} flagOnly />
+            </Button>
+          </React.Fragment>
+        ))}
+      </div>),
+    permission: null
+  });
+
+  return (
+    <div className="EducanduPageTemplate">
+      <header className="EducanduPageTemplate-headerArea">
+        <div className="EducanduPageTemplate-header">
+          <div className="EducanduPageTemplate-headerContent EducanduPageTemplate-headerContent--left">
+            <a href="/" className="EducanduPageTemplate-logoAndName">
+              <EducanduLogo size="small" />&nbsp;&nbsp;educandu
+            </a>
+          </div>
+          <div className="EducanduPageTemplate-headerContent EducanduPageTemplate-headerContent--center">
+            {headerActionComponents}
+          </div>
+          <div className="EducanduPageTemplate-headerContent EducanduPageTemplate-headerContent--right">
+            <div className="EducanduPageTemplate-loginLogoutButton">
+              <LoginLogout />
+            </div>
+            <LinkPopover items={pageMenuItems} trigger="hover" placement="bottomRight">
+              <Button className="EducanduPageTemplate-headerButton" icon={<MenuOutlined />} />
+            </LinkPopover>
+          </div>
+        </div>
+        {!fullScreen && alerts && alerts.map((alert, index) => (
+          <Alert key={index.toString()} message={alert.message} type={alert.type || 'info'} banner />
+        ))}
+      </header>
+      <main className={contentAreaClasses}>
+        <div className={contentClasses}>
+          {children}
+        </div>
+      </main>
+      <footer className="EducanduPageTemplate-footer">
+        <div className="EducanduPageTemplate-footerContent">
+          {(settings?.footerLinks?.[language] || []).map((fl, index) => (
+            <span key={index.toString()} className="EducanduPageTemplate-footerLink">
+              <a href={urls.getArticleUrl(fl.documentSlug)}>{fl.linkTitle}</a>
+            </span>
+          ))}
+        </div>
+      </footer>
+    </div>
+  );
+}
+
+EducanduPageTemplate.propTypes = {
+  alerts: PropTypes.arrayOf(PropTypes.shape({
+    message: PropTypes.node.isRequired,
+    type: PropTypes.oneOf(['success', 'info', 'warning', 'error'])
+  })),
+  children: PropTypes.node,
+  fullScreen: PropTypes.bool,
+  headerActions: PropTypes.arrayOf(PropTypes.shape({
+    handleClick: PropTypes.func.isRequired,
+    icon: PropTypes.elementType.isRequired,
+    key: PropTypes.string.isRequired,
+    permission: PropTypes.string,
+    text: PropTypes.string.isRequired,
+    type: PropTypes.string,
+    loading: PropTypes.bool
+  }))
+};
+
+EducanduPageTemplate.defaultProps = {
+  alerts: [],
+  children: null,
+  fullScreen: false,
+  headerActions: []
+};
+
+export default EducanduPageTemplate;
```

Comment:
Man this is really cool, full control now over the layout

Reply:
I guess it's unfortunately still not enough to completely drive elmu and oma, but a firts step for now. We have to sit together and think of a UI strategy (also including the future merge of doc and edit-doc pages).